beacon-chain/sync/verify/blob.go

```diff
@@ -2,70 +2,42 @@ package verify
 
 import (
 	"github.com/pkg/errors"
-	field_params "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
 	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
 	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
-	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/v4/runtime/version"
 )
 
 var (
 	errBlobVerification          = errors.New("unable to verify blobs")
-	ErrMismatchedBlobBlockRoot   = errors.Wrap(errBlobVerification, "BlockRoot in BlobSidecar does not match the expected root")
-	ErrMismatchedBlobBlockSlot   = errors.Wrap(errBlobVerification, "BlockSlot in BlobSidecar does not match the expected slot")
+	ErrIncorrectBlobIndex        = errors.New("incorrect blob index")
+	ErrBlobBlockMisaligned       = errors.Wrap(errBlobVerification, "root of block header in blob sidecar does not match block root")
 	ErrMismatchedBlobCommitments = errors.Wrap(errBlobVerification, "commitments at given slot, root and index do not match")
-	ErrMismatchedProposerIndex   = errors.Wrap(errBlobVerification, "proposer index does not match")
-	ErrIncorrectBlobIndex        = errors.Wrap(errBlobVerification, "incorrect blob index")
 )
 
 // BlobAlignsWithBlock verifies if the blob aligns with the block.
-func BlobAlignsWithBlock(blob *ethpb.DeprecatedBlobSidecar, block blocks.ROBlock) error {
+func BlobAlignsWithBlock(blob blocks.ROBlob, block blocks.ROBlock) error {
```

Comment:
Note there are some verifications that are missing here and I will address them in my PR